core/src/main/scala/chisel3/Layer.scala

```diff
@@ -28,12 +29,23 @@ object layer {
     case object Bind extends Type
   }
 
+  sealed trait OutputDirBehavior
+  case object DefaultOutputDir extends OutputDirBehavior
+  case object NoOutputDir extends OutputDirBehavior
+  case class CustomOutputDir(val path: Path) extends OutputDirBehavior
```

Comment:
```suggestion
  case class CustomOutputDir(path: Path) extends OutputDirBehavior
```
The `val` is redundant in a case class.